doc/dev/tests.md

```diff
@@ -0,0 +1,378 @@
+* sdist: creates a zipped distribution of our files that the end user could install with pip
+
+## `devtools_testutils` Package
+The Azure SDK team has created some in house tools to help with easier testing. These additional tools are located in the `devtools_testutils` package that was installed with your `dev_requirements.txt`. In this package are the preparers that will be commonly used throughout the repository to test various resources. A preparer is a way to programmatically create fresh resources to run our tests against and then deleting them after running a test suite. These are helpful to help guarantee standardized behavior by starting each test group from a fresh resource and account. For more information on writing and building preparers following this tutorial on how to create the [app-configuration account preparer](https://github.com/Azure/azure-sdk-pr/issues/458).
+
+Also in this package is the `AzureTestCase` object which every test case object should inherit from. This management object takes care of creating and scrubbing recordings to make sure secrets are not added to the recordings files (and subsequently to the git history) and authenticating clients for test methods.
+
+## Writing New Tests
+SDK tests are based on the `scenario_tests` subpackage located in [`azure-sdk-for-python/tools/azure-devtools/src/azure_devtools`](https://pypi.org/project/azure-devtools/). `scenario_tests` is a general, mostly abstracted framework which provides several useful features for writing SDK tests, ie:
+* HTTP interaction recording and playback using [vcrpy](https://pypi.python.org/pypi/vcrpy)
+* Creation and cleanup of helper resources, such as resource groups, storage accounts, etc. which can be used in order to test services
+* Processors for modifying requests and responses when writing or reading recordings (for example, to to avoid recording credential information)
+* Patches for overriding functions and methods that don't work well with testing frameworks (such as long-running operations)
+
+Code in the [`azure-sdk-tools/devtools_testutils`](https://github.com/Azure/azure-sdk-for-python/tree/master/tools/azure-sdk-tools/devtools_testutils) directory provides concrete implementations of the features provided in `scenario_tests` that are oriented around use in SDK testing and that you can use directly in your unit tests.
+
+## Define credentials
+When you run tests in playback mode, they use a fake credentials file, located at `tools/azure-sdk-tools/devtools_testutils/mgmt_settings_fake.py` to simulate authenticating with Azure. The test framework uses the values stored in the `mgmt_settings_real.py` to take care of authenticating clients for you.
+
+In live mode, the credentials need to be real so that the tests are able to connect to the service. Copy the `mgmt_settings_fake.py` file to a new file named `mgmt_settings_real.py` within the same directory. Then make the following changes:
```

Comment:
We should promote .env, and stop talking about mgmt_settings_reals, or in addendum/advanced later